walrus/api.py

```diff
@@ -9,20 +8,52 @@ def hello():
     return "Hello World"
 
 
-def get_capital(country_name, raise_errors=False):
-    # Load the data
-    data = pd.read_csv(file_path)
+def find_capital(country_name, raise_errors=False):
+    # Dictionary to store country and capital as key-value pair
+    country_capital = {}
 
-    # Find the row where the 'CountryName' column matches the input country_name
-    country_data = data[data['CountryName'] == country_name]
+    # Open the csv file with the reading mode
+    with open(file_path, 'r') as file:
+        # Split the file line by line and store it in a list
+        data = file.readlines()
+
+        # Header of the csv file
+        header = data[0].split(",")
+
+        # Indexes for country and capitals, respectively
+        country_col = header.index('CountryName')
+        capital_col = header.index('CapitalName')
+
+        # Ignore the first line
+        # Store country and its capital as a key-value pair
+        for line in data[1:]:
+            values = line.split(",")
+            country_capital[values[country_col]] = values[capital_col]
+
+    capital = country_capital.get(country_name)
 
     # Check if any data was found
-    if country_data.empty:
+    if capital is None:
         if raise_errors:
             raise ValueError(f"No data found for country: {country_name}")
         else:
             return None
 
-    # Extract and return the capital from the matched row
-    return country_data.iloc[0]['CapitalName']
+    # Extract and return the capital
+    return capital
+
+
+def get_capital(country_names, raise_errors=False):
+    # Handle a single country name
+    if isinstance(country_names, str):
+        return find_capital(country_names, raise_errors)
+
+    # Handle multiple country names
+    else:
+        # Dictionary to store the country and capitals of requested countries
+        results = {}
```

Comment:
Same as above - Let's see if we can do the entire else part in a single line of code.

Reply:
Same as above (Appended the entire else part to a single line of code, but separated them into different lines anyway to follow flake8 convention.)

Reply:
Thanks!